sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -494,44 +494,54 @@ object DateTimeUtils {
   }
 
   /**
-   * Returns the month value for the given date. The date is expressed in days
-   * since 1.1.1970. January is month 1.
+   * Split date (expressed in days since 1.1.1970) into four fields:
+   * year, month (Jan is Month 1), dayInMonth, daysToMonthEnd (0 if it's last day of month).
    */
-  def getMonth(date: Int): Int = {
+  def splitDate(date: Int): (Int, Int, Int, Int) = {
```

Comment:
I've [benchmarked](https://gist.github.com/yjshen/2c48cfe6201a97e6e1f5) it.
While comparing `getDayOfMonth` using `splitDate` vs the origin version, no significant performance reduction I think.
While considering the origin version of `MonthBetween` which would call `getMonth`, `getDayOfMonth`, `getLastDayInMonthOfMonth`, we could only call `splitDate` for a single time.